extensions/project-descriptor.md

```diff
@@ -8,80 +8,120 @@ A project descriptor is a file that MAY contain configuration for apps, services
 id = "<string>" # machine readable
 name = "<string>" # human readable
 version = "<string>"
 authors = ["<string>"]
 documentation-url = "<url>"
 source-url = "<url>"
 
-[[project.licenses]]
+[[_.licenses]]
 type = "<string>"
 uri = "<uri>"
 
-[build]
-include = ["<string>"]
-exclude = ["<string>"]
-[[build.buildpacks]]
-id = "<string>"
-version = "<string>"
-uri = "<string>"
-[[build.env]]
-name = "<string>"
-value = "<string>"
-[metadata]
+[_.metadata]
 # additional arbitrary keys allowed
 ```
 
 The following sections describe each part of the schema in detail.
 
-## `[project]`
-
-The top-level `[project]` table MAY contain configuration about the repository, including `id` and `version`. It MAY also include metadata about how it is authored, documented, and version controlled.
+#### `[_]`
 
-The `project.id`
+The top-level `_` table MAY contain configuration about the repository, including `id` and `version`. It MAY also include metadata about how it is authored, documented, and version controlled. It MUST contain `api`  to denote which version of this API the descriptor is using.
 
 ```toml
-[project]
+[_]
+api = "<string>"
 id = "<string>"
 name = "<string>"
 version = "<string>"
+authors = ["<string>"]
+documentation-url = "<url>"
+source-url = "<url>"
 ```
 
+* `api` - version identifier for the schema of the `_` table and structure of the project descriptor file. This version format follows the rules of the [Buildpack API Version](https://github.com/buildpacks/spec/blob/main/buildpack.md#buildpack-api-version).
 * `id` - (optional) the machine readable identifier of the project (ex. "com.example.myservice")
 * `name` - (optional) the human readable name of the project (ex. "My Example Service")
 * `version` - (optional) and arbitrary string representing the version of the project
 * `authors` - (optional) the names and/or email addresses of the project's authors
 * `documentation-url` - (optional) a URL to the documentation for the project
 * `source-url` - (optional) a URL to the source code for the project
 
-## `[[project.licenses]]`
+#### `[[_.licenses]]`
 
 An optional list of project licenses.
 
 * `type` - This MAY use the [SPDX 2.1 license expression](https://spdx.org/spdx-specification-21-web-version), but is not limited to identifiers in the [SPDX Licenses List](https://spdx.org/licenses/).
 * `uri` - If this project is using a nonstandard license, then this key MAY be specified in lieu of or in addition to `type` to point to the license.
 
-## `[build.include]` and `[build.exclude]`
+
+#### `[_.metadata]`
+
+Keys in this table are not validated and can be used by users.
+
+```toml
+[_.metadata.foo]
+checksum = "a28a0d7772df1f918da2b1102da4ff35"
+```
+
+### Non-`[_]` Tables
+All other tables besides `_` will use reverse domains, i.e. buildpacks.io will be `[io.buildpacks]`. These tables can be optionally versioned with a schema version API number using the `api` field.
+
+### `[io.buildpacks]`
+
+API Version: `0.1`
```

Comment:
How does having two separately defined API versions interact with our versioning of this spec? Would we ever release a version `0.2` of this table without change the API version of the project file? If so, how do we want to handle that process?

Reply:
Unsure, I'm happy to combine them if we want, but they technically are separate tables. I had them all in one file for the first pass.